src/Orleans.Core/CodeGeneration/TypeUtils.cs

```diff
@@ -355,7 +355,9 @@ public static string GetFullName(Type t)
                        + new string(',', t.GetArrayRank() - 1)
                        + "]";
             }
-            return t.FullName ?? (t.IsGenericParameter ? t.Name : t.Namespace + "." + t.Name);
+
+            if (t.FullName != null) return t.FullName;
```

Comment:
Why change this? It's more code and less efficient and gives the same result.

Reply:
It wont be in here when I take the WIP tag off, I likely just did it to make debugging easier at some point.